libs/cardano-protocol-tpraos/src/Cardano/Protocol/TPraos/BHeader.hs

```diff
@@ -358,12 +357,8 @@ prevHashToNonce = \case
 issuerIDfromBHBody :: CC.Crypto crypto => BHBody crypto -> KeyHash 'BlockIssuer crypto
 issuerIDfromBHBody = hashKey . bheaderVk
 
-bHeaderSize ::
-  forall crypto.
-  (CC.Crypto crypto) =>
-  BHeader crypto ->
-  Int
-bHeaderSize = BS.length . serialize'
+bHeaderSize :: forall crypto. BHeader crypto -> Int
+bHeaderSize = fromIntegral . BSL.length . bHeaderBytes
```

Comment:
`fromIntegral` is redundant
```suggestion
bHeaderSize = BSL.length . bHeaderBytes
```

Reply:
@lehins and I chatted about this, turns out `BSL.length` returns an `Int64`. We are working under the assumption that we do not support 32 bit architecture so that this conversion make sense.

Reply:
This can only be a problem if a Block header serializes to a ByteString larger than 2Gb on a 32bit machine, which is definitely not a realistic scenario. Moreover previous implementation had the same semantics, except it first converted the lazy ByteString into a strict one.

This comment was because I totally forgot that length on lazy ByteString returns `Int64` and not `Int`